### DISCUSSION TITLE
Shelly pro 3em

### DISCUSSION THREAD

**Discussion by Heinz7000** [fabian-lauer/dbus-shelly-3em-smartmeter#42] [Show and tell]
Hallo,
wäre es möglich den pro auch zu implementieren?

lg Heinz

**Comment by fgellhorn** [comment]
Hallo Heinz, 

ich bin heut ebenfalls an der Integration des Pro EM gescheitert...da haben sich leider die Endpunkte für die Datenabfrage geändert...ich habe das 'dbus-shelly-3em-smartmeter.py' Script entsprechend angepasst, sodass es jetzt mit dem Pro läuft.
Ich kann das gern zur Verfügung stellen...müsste nur die Tage bei Gelegenheit nochmal drüber schauen und aufräumen

![Bildschirmfoto 2023-04-02 um 21 11 02](https://user-images.githubusercontent.com/30745012/229373857-5ce06abf-dfbc-4b8b-9b05-9221474d5315.jpg)

**Comment by Heinz7000** [comment]
Hallo,
das hört sich super an. Freu mich schon auf das Script.
Lg Heinz

**Comment by fgellhorn** [comment]
Alternativ wäre hier der Treiber direkt von Victron: https://github.com/victronenergy/dbus-shelly

**Comment by Heinz7000** [comment]
Hallo, Danke, ich schaue mal, aber das ist schon komplizierter wie mit dem fertigen Script🙄

**Comment by fgellhorn** [comment]
Hallo Heinz, ich habe mal ein wenig aufgeräumt und hoffe, dass ich es nicht kaputt optimiert habe^^
Mein angepasstes Script hänge ich mal an.

Gutes Gelingen :)
Frank
[dbus-shelly-3em-smartmeter.txt](https://github.com/fabian-lauer/dbus-shelly-3em-smartmeter/files/11140876/dbus-shelly-3em-smartmeter.txt)

**Comment by Heinz7000** [comment]
Hallo, super vielen Dank, es läuft 😋

**Comment by axbee90** [comment]
Hallo zusammen, 
wie bekomme ich das angepasste Script jetzt auf den Raspy? Was muss ich da selbst noch anpassen? Bin leider noch etwas grün hinter den Ohren, was Putty und Co angeht.

**Comment by Heinz7000** [comment]
Hallo, du musst nach der hier beschriebenen Installation den Inhalt der dbus-shelly-3em-smartmeter.py gegen den Inhalt der Textdatei tauschen. Dann rebelt und es sollte klappen.

**Comment by fabian-lauer** [comment]
Hey Leute - danke für euren Einsatz und die Erweiterung.
Würde dieses Thema mal hier schließen (ist glaube kein Issue) und in den Bereich der Diskussionen verschieben.

Gruß
Fabian

**Reply by Patrick38110** [reply]
Hallo Fabian, erstmal Herzlichen Dank für dein Engagement. Ist in der aktuellen Version der Shelly Pro 3EM unterstützt oder ist hier der Treiber von Victron [](https://github.com/victronenergy/dbus-shellyl) zu nutzen? Bin etwas verunsichert was ich jetzt nutzen soll.

**Comment by fuxe13** [comment]
Hallo!

Möchte mich anschliessen....habe ebenfalls den Shelly Pro 3em. Hab diesen Script einkopiert und geht soweit.

Habe auch die OpenDTU installiert. Jetzt wird der AC-Eingang und die AC-Last addiert. Somt ist nicht mehr sichtbar, was auf welcher "Phase" eingespeist wird... 
![victron01](https://github.com/fabian-lauer/dbus-shelly-3em-smartmeter/assets/121746821/6a76159a-42fe-459d-ba00-41a4e44d1282)


Was müsste man wo anpassen??

**Comment by fellek** [comment]
Mit der Version Venus OS 3.00 ist ein Script **dbus-shelly** aufgetaucht. In der Beschreibung steht, dass es auf den shelly 3em pro ausgerichtet ist: 

[https://github.com/victronenergy/dbus-shelly](https://github.com/victronenergy/dbus-shelly)

Leider habe ich keinen shelly 3em pro, sodass ich das nicht ausprobieren kann.

**Reply by Patrick38110** [reply]
Hallo zusammen, hat es schon jemand mit der Shelly Pro Version getestet?

**Comment by Stobli** [comment]
Ich hab mir das ganze mal auf meinen Respi 4 installiert und das Script ausgeführt. Dann die Config zu meinen 3EM Pro angepasst aber er findet ihn nicht. User & PW habe ich keines vergeben.
Muss im Venus os was extra eingestellt werden ?

**Comment by fellek** [comment]
Hallo @Stobli, falls noch nicht geschehen muss auf der Weboberfläche von Venus OS der Wert unter 
```
Einstellungen > ESS > Netzmessung = Externer Zähler
```
eingestellt werden. 

Im Zweifel Venus OS neustarten.

Hoffe es hilft dir weiter.

**Comment by kufo78** [comment]
Bitte für den Shelly Pro 3 EM dieses Script verwenden

https://github.com/funkmaster86/dbus-shelly-pro-3em-smartmeter

Läuft dann auch

**Comment by Robman1** [comment]
Wie kann man das Skript wieder "deinstallieren"? 
Einfach das Verzeichnis "/data/dbus-shelly-3em-smartmeter" löschen?
Danke

**Reply by fellek** [reply]
Am besten `./uninstall.sh` ausführen. erst danach den Ordner löschen, wenn du ihn nicht mehr brauchst.
